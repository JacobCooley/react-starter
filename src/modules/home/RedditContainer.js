import { connect } from 'react-redux';
import RedditComponent from './RedditComponent';

import { homeOperations } from './duck';

const mapStateToProps = (state) => {
    const { subredditData, showRedditSpinner } = state.home;
    return {
        subredditData,
        showRedditSpinner
    }
};

const mapDispatchToProps = (dispatch) => {
    const fetchSubredditJson = (subreddit) => {
        dispatch(homeOperations.fetchSubredditJson(subreddit))
    };
    return { fetchSubredditJson };
};

const RedditContainer = connect(
    mapStateToProps,
    mapDispatchToProps
)(RedditComponent);

export default RedditContainer;