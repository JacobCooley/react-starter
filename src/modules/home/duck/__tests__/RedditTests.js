import RedditComponent from '@home/RedditComponent'
import RedditContainer from '@home/RedditContainer'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'

configure({ adapter: new Adapter() })

describe('<RedditComponent /> --- Snapshot',()=>{
	it('+++capturing Snapshot of <RedditComponent />', () => {
		const renderedValue =  renderer.create(<RedditComponent subredditData={[]}/>).toJSON()
		expect(renderedValue).toMatchSnapshot();
	});
});

describe('<RedditComponent />', () => {
	let wrapper
	const subredditData = []
	
	beforeEach(() => {
		wrapper = shallow(<RedditComponent subredditData={subredditData}/>)
		
	})
	
	it('+++ render the DUMB component', () => {
		expect(wrapper.length).toEqual(1)
	})
	
	it('+++ contains subredditData', () => {
		expect(wrapper.find('ul').prop('children')[0]).toEqual(subredditData)
	})
})

describe('<RedditContainer />>', () => {
	const initialState = {
		home: {
			subreddit: '',
			showSpinner: false,
			subredditData: []
		}
	}
	const mockStore = configureStore()
	let store, container
	
	beforeEach(() => {
		store = mockStore(initialState)
		container = shallow(<RedditContainer store={store}/>)
	})
	
	it('+++ render the connected(SMART) component', () => {
		expect(container.length).toEqual(1)
	})
	
	it('+++ check Prop matches with initialState', () => {
		expect(container.prop('subredditData')).toEqual(initialState.home.subredditData)
	})
	
})