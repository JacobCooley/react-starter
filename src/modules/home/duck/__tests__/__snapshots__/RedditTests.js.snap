// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RedditComponent /> --- Snapshot +++capturing Snapshot of <RedditComponent /> 1`] = `
<ul>
  <button
    onClick={[Function]}
  >
    Show ReactJS subreddits
  </button>
</ul>
`;
